src/vcpkg/base/files.cpp

```diff
@@ -825,8 +825,8 @@ namespace
             if (!S_ISDIR(s.st_mode))
             {
                 // if it isn't still a directory something is racy
-                ec = std::make_error_code(std::errc::device_or_resource_busy);
```

Comment:
I'm pretty sure this is now incorrect because we aren't marking the recursive error as `device_or_resource_busy` any longer.

Reply:
(That is to say, mark_recursive_error excepts `ec` to contain the error already when it is called IIRC)

Reply:
I moved the assignment below `mark_recursive_error` because `mark_recursive_error()` overwrites ec.

Reply:
```cpp
    void mark_recursive_error(const Path& base, std::error_code& ec, Path& failure_point)
    {
        failure_point = base;
        ec.assign(errno, std::generic_category());
    }
```

Reply:
(which means I should probably change this to not call mark_recursive_error at all)